bin/yb-ctl

```diff
@@ -1331,20 +1447,49 @@ class ClusterControl:
         info_kv_list = []
         if not tserver_daemon_id:
             return info_kv_list
+
+        tserver_ip_address = tserver_daemon_id.get_ip_address()
+
+        # -----------------------------------------------------------------------------------------
+        # ysqlsh command line
+        # -----------------------------------------------------------------------------------------
+
         if self.is_ysql_enabled():
-            ysql_port = self.options.base_ports[tserver_daemon_id.daemon_type]["ysql_rpc"]
+            ysql_port = self.options.get_ysql_port()
```

Comment:
the `options.get_xxx_port` only seem to be called once, from here, so I'd suggest not having them and just have the inlined code fetching from the dict